scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -666,3 +667,256 @@ def constr_f(x):
                   1.0, False, np.array([1., 0.50])))
         assert (fn(1.0, False, np.array([0.5, 0.5]),
                   1.0, False, np.array([1., 0.4])) == False)
+
+    def test_L1(self):
+        # Lampinen ([5]) test problem 1
+
+        def f(x):
+            x = np.hstack(([0], x))  # 1-indexed to match reference
+            fun = np.sum(5*x[1:5]) - 5*x[1:5]@x[1:5] - np.sum(x[5:])
+            return fun
+
+        A = np.zeros((10, 14))  # 1-indexed to match reference
+        A[1, [1, 2, 10, 11]] = 2, 2, 1, 1
+        A[2, [1, 10]] = -8, 1
+        A[3, [4, 5, 10]] = -2, -1, 1
+        A[4, [1, 3, 10, 11]] = 2, 2, 1, 1
+        A[5, [2, 11]] = -8, 1
+        A[6, [6, 7, 11]] = -2, -1, 1
+        A[7, [2, 3, 11, 12]] = 2, 2, 1, 1
+        A[8, [3, 12]] = -8, 1
+        A[9, [8, 9, 12]] = -2, -1, 1
+        A = A[1:, 1:]
+
+        b = np.array([10, 0, 0, 10, 0, 0, 10, 0, 0])
+
+        L = LinearConstraint(A, -np.inf, b)
+
+        bounds = [(0, 1)]*9 + [(0, 100)]*3 + [(0, 1)]
+
+        # using parameters from [5]
+        res = differential_evolution(f, bounds, maxiter=4000, popsize=20,
```

Comment:
If following Lampinen exactly is important then `updating='deferred'` probably needs to be used. Otherwise I'd stick to the default `differential_evolution` keyword arguments.

Reply:
I don't think it's important. I only attempted to follow Lampinen's parameters because I wasn't observing the same accuracy with default parameters. It didn't get much better after setting the parameters, so I think it's fine to use defaults.

Reply:
defaults would go faster.